cli/src/torii/routing.rs

```diff
@@ -273,40 +273,28 @@ async fn handle_blocks_stream(sumeragi: Arc<Sumeragi>, mut stream: WebSocket) ->
         .await?;
 
     loop {
-        let blocks = sumeragi.blocks_from_height(from_height.try_into().wrap_err("Failed to convert `from_height` into `usize`. You should consider upgrading to a 64-bit arch")?);
-        if blocks.is_empty() {
-            break;
+        match kura.get_block_by_height(from_height) {
+            None => break,
+            Some(block) => {
+                stream
+                    .send(VersionedBlockPublisherMessage::from(
+                        BlockPublisherMessage::from(VersionedCommittedBlock::clone(&block)), // TODO: Remove unnecessary clone.
+                    ))
+                    .await?;
+
+                let message: VersionedBlockSubscriberMessage = stream.recv().await?;
+                if let BlockSubscriberMessage::BlockReceived = message.into_v1() {
+                    from_height += 1;
+                } else {
+                    return Err(eyre!("Expected `BlockReceived` message"));
```

Comment:
Rewrite `if let` as a match. Match onto `BlockReceived` in one block, name the other `message_as_v1` and 

```suggestion
                    eyre::bail!("Expected `BlockReceived` message, got {}", message_as_v1)));
```

Reply:
We need as much information as possible. Given that errors shouldn't happen frequently, adding a bit of extra info doesn't hurt.